gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/tasks/CreateManifestTask.java

```diff
@@ -427,7 +427,7 @@ private void ensureLockfileIsUpToDate(List<ProductDependency> productDeps) {
                 .forEach(productDependency -> discoveredProductDependencies.merge(
                         new ProductId(productDependency.getProductGroup(), productDependency.getProductName()),
                         productDependency,
-                        (key, oldValue) -> ProductDependencyMerger.merge(oldValue, productDependency)));
```

Comment:
oof that's subtle.  I wonder if error-prone should have told us `key` was unused and complained?